src/finch/compiled.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+from .julia import jl
+from .tensor import Tensor
+
+
+def compiled(func):
+    @wraps(func)
+    def wrapper_func(*args, **kwargs):
+        new_args = []
+        for arg in args:
+            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
+                new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
+            else:
+                new_args.append(arg)
+
+        result = func(*new_args, **kwargs)
+        result_tensor = Tensor(jl.Finch.compute(result._obj))
+
+        return result_tensor
+
+    return wrapper_func
+
+
+def to_lazy(tensor: Tensor):
+    if not jl.isa(tensor._obj, jl.Finch.LazyTensor):
+        return Tensor(jl.Finch.LazyTensor(tensor._obj))
+    else:
+        tensor
+
+
+def compute(tensor: Tensor):
+    if not jl.isa(tensor._obj, jl.Finch.LazyTensor):
+        raise ValueError("Input tensor must be a LazyTensor")
```

Comment:
```suggestion
        return tensor
```

A non-lazy `Tensor` is already computed.

Reply:
Done!